saplings/circuits/src/components/circuitDetails/CircuitDetails.js

```diff
@@ -0,0 +1,165 @@
+          </Link>
+        </div>
+        <div className="circuits-title">
+          <h4>
+            Circuit
+            {circuitId}
+          </h4>
+          <div className="managementType">
+            Gameroom
+            <span>
+              <FontAwesomeIcon icon={faQuestionCircle} />
+            </span>
+          </div>
+        </div>
+      </div>
+      <div className="main-content">
+        <div className="midContent">
+          <div className="circuit-stats">
+            <div className="stat total-circuits">
+              <span className="stat-count circuits-count">
+                {circuit.members.length}
+              </span>
+              Nodes
+            </div>
+            <div className="stat action-required">
+              <span className="stat-count action-required-count">
+                {circuit.roster.length}
+              </span>
+              Services
+            </div>
+          </div>
+        </div>
+
+        <NodesTable circuit={circuit} nodes={circuit.members} />
+      </div>
+    </div>
+  );
+};
+
+const NodesTable = ({ circuit, nodes }) => {
+  const [fullNodes, setNodes] = React.useState(null);
+
+  React.useEffect(() => {
+    const fetchNodes = async () => {
+      try {
+        const apiNodes = await getNodeRegistry();
+        const filteredNodes = apiNodes.filter(
+          node => !!nodes.find(id => id === node.identity)
+        );
+        setNodes(filteredNodes);
+      } catch (e) {
+        throw Error(`Unable to fetch nodes from the node registry: ${e}`);
+      }
+    };
+
+    fetchNodes();
+  }, [nodes]);
+
+  if (fullNodes === null) {
+    return <div />;
+  }
+
+  let rows = [
+    <tr>
+      <td colSpan="5" className="no-nodes-msg">
+        No Nodes found for this circuit
+      </td>
+    </tr>
+  ];
+
+  if (fullNodes.length > 0) {
+    rows = fullNodes.map(node => {
+      return (
+        <tr className="table-row">
+          <td>{node.identity}</td>
+          <td>{node.displayName}</td>
+          <td>
+            {node.metadata.company || node.metadata.organization || 'N/A'}
+          </td>
+          <td>{node.endpoints[0] || 'N/A'}</td>
+          <td>
+            <NodeStatus circuit={circuit} nodeId={node.identity} />
+          </td>
+        </tr>
+      );
+    });
+  }
+
+  return (
+    <div className="table-container">
+      <table className="nodes-table">
+        <tr className="table-header">
+          <th>Name</th>
+          <th>Alias</th>
+          <th>Company</th>
+          <th>Endpoint</th>
+          <th>Status</th>
+        </tr>
+        {rows}
+      </table>
+    </div>
+  );
+};
+
+NodesTable.propTypes = {
+  circuit: PropTypes.arrayOf(Circuit).isRequired,
+  nodes: PropTypes.arrayOf(Object).isRequired
+};
+
+const NodeStatus = ({ circuit, nodeId }) => {
+  if (circuit.awaitingApproval() && !circuit.nodeHasVoted(nodeId)) {
+    return <span className="status awaiting-approval">Awaiting approval</span>;
```

Comment:
The rules for the classes `status` and `awaiting-approval` don't seen to be being applied here

Reply:
I will probably deal with those when working on the status tags.